light-jobs-test/src/main/java/com/hazelcast/jet/tests/lightjobs/LightJobsTest.java

```diff
@@ -0,0 +1,180 @@
+    private static final int DEFAULT_LOG_JOB_COUNT_THRESHOLD = 1000;
+    private static final int DEFAULT_SLEEP_BETWEEN_JOBS_MS = 100;
+
+    private IMap<String, String> sourceBatchMap;
+    private IMap<String, String> sourceJournalMap;
+    private IMap<String, String> sourceIncorrectMap;
+
+    private int itemCount;
+    private int sleepAfterJobMillis;
+    private int logJobCountThreshold;
+
+    public static void main(String[] args) throws Exception {
+        new LightJobsTest().run(args);
+    }
+
+    @Override
+    protected void init(HazelcastInstance client) throws Exception {
+        sourceBatchMap = client.getMap(SOURCE_BATCH_MAP_NAME);
+        sourceBatchMap.clear();
+        sourceJournalMap = client.getMap(SOURCE_JOURNAL_MAP_NAME);
+        sourceJournalMap.clear();
+        sourceIncorrectMap = client.getMap(SOURCE_INCORRECT_MAP_NAME);
+        sourceIncorrectMap.clear();
+
+        itemCount = propertyInt("itemCount", DEFAULT_ITEM_COUNT);
+        sleepAfterJobMillis = propertyInt("sleepAfterJobMillis", DEFAULT_SLEEP_BETWEEN_JOBS_MS);
+        logJobCountThreshold = propertyInt("logJobCountThreshold", DEFAULT_LOG_JOB_COUNT_THRESHOLD);
+
+        for (int i = 0; i < itemCount; i++) {
+            String item = Integer.toString(i);
+            sourceBatchMap.put(item, item);
+            sourceJournalMap.put(item, item);
+        }
+        sourceIncorrectMap.put("ignore", "ignore");
+    }
+
+    @Override
+    protected void test(HazelcastInstance client, String name) {
+        long begin = System.currentTimeMillis();
+        long jobCount = 0;
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+            runBatchJob(client, jobCount);
+            sleepMillis(sleepAfterJobMillis);
+            runStreamJob(client, jobCount);
+            sleepMillis(sleepAfterJobMillis);
+            runIncorrectJob(client);
+            sleepMillis(sleepAfterJobMillis);
+
+            if (jobCount % logJobCountThreshold == 0) {
+                logger.info("Job count: " + jobCount);
+            }
+            jobCount++;
+        }
+        logger.info("Final job count: " + jobCount);
+    }
+
+    private void runBatchJob(HazelcastInstance client, long jobCount) {
+        String outputName = OUTPUT_BATCH_MAP_PREFIX + jobCount;
+        IMap<String, String> map = client.getMap(outputName);
+        assertTrue(map.isEmpty());
+
+        client.getJet().newLightJob(batchPipeline(outputName)).join();
+        assertEquals(itemCount, map.size());
+
+        map.destroy();
+    }
+
+    private void runStreamJob(HazelcastInstance client, long jobCount) {
+        String outputName = OUTPUT_STREAM_MAP_PREFIX + jobCount;
+        IMap<String, String> map = client.getMap(outputName);
+        assertTrue(map.isEmpty());
+
+        Job job = client.getJet().newLightJob(streamPipeline(outputName));
+        waitForMapSize(map, 20, 1000);
+        job.cancel();
+        assertEquals(itemCount, map.size());
```

Comment:
we cannot be sure that we cancelled at exactly right time to assert size to itemCount. 
`assertTrue(size > itemCount)`